src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,263 @@
+      -d        Local path to files for achiving/compressing/size checking
+      -s        Enable running cart_logtest.py
+      -v        Display commands being executed
+      -h        Display this help and exit
+"
+}
+
+#######################################
+# CONSTANT variables.
+#######################################
+CART_TESTLOG="./cart/util/cart_logtest.py"
+
+#######################################
+# Print set GLOBAL variables.
+#######################################
+display_set_vars() {
+    echo "Set Variables:
+    COMPRESS = ${COMPRESS}
+    THRESHOLD = ${THRESHOLD}
+    REMOTE_DEST = ${REMOTE_DEST}
+    LOCAL_SRC = ${LOCAL_SRC}
+    CART_LOGTEST = ${CART_LOGTEST}
+    VERBOSE = ${VERBOSE}"
+}
+
+#######################################
+# Convert human readable size values to respective byte value.
+# Arguments:
+#   size: size value to convert to bytes. i.e. 5mb, 40gb, 6k, 10g
+# Returns:
+#   converted byte value.
+#######################################
+human_to_bytes() {
+    h_size=${1,,}
+
+    declare -A units=(
+        [b]=1
+        [kb]=$(python -c "print (2**10)")
+        [k]=$(python -c "print (2**10)")
+        [mb]=$(python -c "print (2**20)")
+        [m]=$(python -c "print (2**20)")
+        [gb]=$(python -c "print (2**30)")
+        [g]=$(python -c "print (2**30)")
+    )
+    if [[ "${h_size,,}" =~ ([0-9.]+)([kbmgt]+) ]]; then
+        val="${BASH_REMATCH[1]}"
+        unit="${BASH_REMATCH[2]}"
+        (( val = val*${units[${unit}]} ))
+        echo "${val}"
+    else
+        echo "Invalid value provided: ${1}"
+        return 1
+    fi
+}
+
+#######################################
+# List and tag files with a 'Y' or 'N' to classify
+# big files based on the threshold provided.
+# Arguments:
+#   logs_dir: path to local logs
+#   threshold: value to use as threhold to classify big files
+# Returns:
+#   stdout output listing the files prepended with a 'Y' or 'N' tag
+#######################################
+list_tag_files() {
+    # shellcheck disable=SC2016,SC1004
+    awk_cmd='{ \
+    if (th <= $1){ \
+        print "Y:",$1,$2} \
+    else { \
+        print "N:",$1,$2}\
+    }'
+
+    # Convert to bytes and run
+    bts=$(human_to_bytes "${2}")
+    du -ab -d 1 ${1} | awk -v th="${bts}" "${awk_cmd}"
+}
+
+check_hw() {
+    dmesg | grep "Hypervisor detected"
+    return $?
+}
+
+compress_files() {
+    find ${1} -maxdepth 0 -type f -size +1M -print0 | xargs -r0 lbzip2 -v
+}
+
+scp_files() {
+    set -ux
+    rc=0
+    copied=()
+    for file in ${1}
+    do
+        ls -sh "${file}"
+        if scp -r "${file}" "${2}"/"$(hostname -s)"-"${file##*/}"; then
+            copied+=("${file}")
+            if ! sudo rm -fr "${file}"; then
+                ((rc++))
+                ls -al "${file}"
+            fi
+        fi
+    done
+    echo Copied "${copied[@]:-no files}"
+    exit "$rc"
+}
+
+get_cartlogtest_files() {
+    set -ux
+    for file in $(ls -d ${1})
```

Comment:
(lint) Iterating over ls output is fragile. Use globs. [SC2045]
(lint) Double quote to prevent globbing and word splitting. [SC2086]